test/AutocompleteBadges.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+import { async } from 'metal';
+import AutocompleteBadges from '../src/AutocompleteBadges';
+import dom from 'metal-dom';
+
+var component;
+var input;
+
+var elements = ['Alabama', 'Arizona', 'California', 'Colorado', 'Florida', 'Indiana'];
+
+describe('AutocompleteBadges', function() {
+
+	afterEach(function() {
+		if (component) {
+			component.dispose();
+		}		
+	});
+
+	it('should check display component', function() {
+		component = new AutocompleteBadges({
+			elements: elements
+		});
+
+		assert.ok(component.visible);
+	});
+
+	it('should check elements length greater than zero with valid query', function() {
+		component = new AutocompleteBadges({
+			elements: elements
+		});
+
+		assert.ok(component.getFilteredElements_('a').length > 0);
+	});
+
+	it('should check elements length equals zero with invalid query', function() {
+		component = new AutocompleteBadges({
+			elements: elements
+		});
+
+		assert.ok(component.getFilteredElements_('sadaswqee').length === 0);
+	});
+
+	it('should process valid query', function(done) {
+		component = new AutocompleteBadges({
+			elements: elements
+		});
+
+		component.autocomplete_.request('a').then(function() {
```

Comment:
It may be useful to expose this autocomplete instance to developers, and it would also help access it in the tests without having to use this private variable. Can you add a simple `getAutocomplete` function that just returns the instance, and change this and other calls in this file to use it instead?